### DISCUSSION TITLE
How i can use my own Cloudflare gateway?

### DISCUSSION THREAD

**Discussion by Delkaloop** [trinib/Adguard-Wireguard-Unbound-Cloudflare#16] [Q&A]
Hi everyone.

Cloudflare Teams offers a customizable DoH-DoT address.
More info: https://scotthelme.co.uk/supercharging-your-dns-with-cloudflare/

I think these are better than Cloudflare original addresses. How i can change it with this set-up? Im using Adguard-Unbound with Stubby and Cloudflared on Raspberry Pi.
What should I change and how should I change it?

**Comment by trinib** [comment]
will look into this

**Reply by Delkaloop** [reply]
These addresses are free for up to 50 users. The good thing is that when these gateway used on Raspberry Pi, it only detects 1 user, no matter how many people use it on the router.

**Comment by trinib** [comment]
@Delkaloop  I signed up on https://dash.teams.cloudflare.com/ with free account and followed these guides from [cloudflare docs](https://developers.cloudflare.com/) :

https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/locations/configuring-a-location/

https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/agentless/

It works with pi set up 👍

I replaced with my gateway in :

### unbound.conf 
```
forward-addr: 172.64.36.1@853#xxxxxxxxx.cloudflare-gateway.com
forward-addr: 172.64.36.2@853#xxxxxxxxx.cloudflare-gateway.com
forward-addr: 2a06:98c1:54::xxxx#xxxxxxxxx.cloudflare-gateway.com
```
![unbound (1)](https://user-images.githubusercontent.com/18756975/157354512-ea70b6ea-24c7-42a0-bc6e-fb11b45c2576.png)
 
### cloudflared file
```
https://YOUR_UNIQUE_SUBDOMAIN.cloudflare-gateway.com/dns-query
```
![cloudflared](https://user-images.githubusercontent.com/18756975/157354520-011b0df9-515f-49e5-bb54-1f35663a26f8.png)

**Comment by trinib** [comment]
> I think these are better than Cloudflare original addresses. How i can change it with this set-up? Im using Adguard-Unbound with Stubby and Cloudflared on Raspberry Pi. What should I change and how should I change it?

@Delkaloop Keep in mind **Cloudflare Zero Trust** is a company/team/across-the-world service of cloudflare for cloud systems and applications

I do not see much benefits from this as a home user when using cloudflare lightweight daemon service and creating a [tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/) (outbound-only connections to Cloudflare’s [edge](https://www.cloudflare.com/learning/serverless/glossary/what-is-edge-computing/)) with pi/linux environment and using adguard to block malware, ransomware, phishing etc and its built in dns proxy to resolve multiple upstream dns servers(DoH&DoT).

# 

# From Cloudflare Docs :

### Cloudflare Zero Trust : 

Cloudflare Zero Trust replaces legacy security perimeters with our global edge, making the Internet faster and safer for teams around the world.

- Zero Trust access for all of your applications.
-  A Secure Web Gateway to protect users and devices.
-  A fast and reliable solution for remote browsing.
-   Enforce consistent access controls across cloud, on-premise and SaaS applications.
-   Integrate with one or more existing identity providers.
-   Protect applications with identity, posture, and context-driven rules.

Zero Trust is a shift of network defenses toward a more comprehensive IT security model that allows organizations to restrict access controls to networks, applications, and environment without sacrificing performance and user experience. In short, a Zero Trust approach trusts no one

 What is the concept of Zero Trust? :
- user/application authentication, device authentication, and trust.

### Cloudflare Gateway : 

Cloudflare Gateway’s DNS resolver introduces security into this flow. Instead of allowing all DNS queries, 
Gateway first checks the hostname being queried against the intelligence Cloudflare has about threats on the Internet. 
If that query matches a known threat, or is requesting a blocked domain configured by an administrator as part of a Gateway policy, 
Gateway stops it before the site could load for the user - and potentially execute code or phish that team member

Cloudflare's secure web gateway keeps your data safe from malware, ransomware, phishing, command & control, Shadow IT, and other Internet risks over all ports and protocols. Log every user interaction with rich details.

Instead of a VPN, users connect to corporate resources through a client or a web browser. As requests are routed and accelerated through Cloudflare's edge, they are evaluated against Zero Trust rules incorporating signals from your identity providers, devices, and other context

###  More from cloudflare blog :

https://blog.cloudflare.com/the-zero-trust-platform-built-for-speed/
 
https://blog.cloudflare.com/gateway-swg/
